packages/usa-date-picker/src/index.js

```diff
@@ -80,30 +80,45 @@ const CALENDAR_YEAR_FOCUSED = `.${CALENDAR_YEAR_FOCUSED_CLASS}`;
 
 const VALIDATION_MESSAGE = "Please enter a valid date";
 
-const MONTH_LABELS = [
-  "January",
-  "February",
-  "March",
-  "April",
-  "May",
-  "June",
-  "July",
-  "August",
-  "September",
-  "October",
-  "November",
-  "December",
-];
-
-const DAY_OF_WEEK_LABELS = [
-  "Sunday",
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-];
+const MONTH_DATE_SEED = Array.from({ length: 12 }).map(
+  (_, i) => new Date(0, i)
+);
+
+const DAY_OF_WEEK_DATE_SEED = Array.from({ length: 7 }).map(
+  (_, i) => new Date(0, 0, i)
+);
+
+let monthLabels;
+let dayOfWeeklabels;
+let dayOfWeeksAbv;
+
+function setCalendarLabels() {
+  const lang = document.documentElement.lang || "en";
+
+  monthLabels = MONTH_DATE_SEED.map((date) =>
+    date.toLocaleString(lang, { month: "long" })
+  );
+
+  dayOfWeeklabels = DAY_OF_WEEK_DATE_SEED.map((date) =>
+    date.toLocaleString(lang, {
+      weekday: "long",
+    })
+  );
+
+  dayOfWeeksAbv = DAY_OF_WEEK_DATE_SEED.map((date) =>
+    date.toLocaleString(lang, {
+      weekday: "narrow",
+    })
+  );
+}
+
+setCalendarLabels();
+
+// NOTE: This event is NOT triggered when setting the document language programatically
+// If the user manually sets document.documentElement.lang they will need to manually dispatch this event
+window.addEventListener("languagechange", () => {
+  setCalendarLabels();
+});
```

Comment:
another option could also be to generate the months and days of the week when rendering the calendar itself. Would need to add a cache to avoid calculating the lists multiple times.